File: M/1323-maximum-69-number/main.go
package main

import "fmt"

func maximum69Number(num int) int {
	var buf []int

	for num > 0 {
		n := num % 10
		buf = append(buf, n)
		num /= 10
	}
	s, e := 0, len(buf)-1
	for s < e {
		buf[s], buf[e] = buf[e], buf[s]
		s++
		e--
	}
	for i := 0; i < len(buf); i++ {
		if buf[i] == 6 {
			buf[i] = 9
			break
		}
	}
	ret := 0
	for i := 0; i < len(buf); i++ {
		ret = ret*10 + buf[i]
	}
	return ret
}

func main() {
	fmt.Println("a")
}

File: M/1323-maximum-69-number/main.rs
impl Solution {
  pub fn maximum69_number(num: i32) -> i32 {
    let mut buf: Vec<i32> = vec![];
    let mut n = num;
    while n > 0 {
      buf.push(n % 10);
      n /= 10;
    }

    for i in (0..buf.len()).rev() {
      if buf[i] == 6 {
        buf[i] = 9;
        break;
      }
    }

    let mut ans: i32 = 0;
    for i in (0..buf.len()).rev() {
      ans = ans * 10 + buf[i];
    }

    ans
  }
}
